benchopt/utils/datasets/simulated.py

```diff
@@ -0,0 +1,57 @@
+"""Helper to generate simulated data in benchopt."""
+import numpy as np
+
+from ..checkers import check_random_state
+
+
+def gen_correlated_data(n_samples=100, n_features=50, rho=0.6,
+                        random_state=None):
+    """Generate correlated design matrix with decaying correlation rho**|i-j|.
+
+    The data are generated using an AR model with reason corr and innovation
+    $\\sigma^2 = 1 - \\rho^2$,
+    $$
+        x_{i+1} = \\rho x_i + \\epsilon \\quad
+        where \\quad \\epsilon \\sim \\mathcal N(0, \\sigma^2)
+    $$
+    The generated features have mean 0, variance 1 and the expected correlation
+    structure
+    $$
+        \\mathbb E[x_i] = 0~, \\quad \\mathbb E[x_i^2] = 1  \\quad
+        and \\quad \\mathbb E[x_ix_j] = \\rho^{|i-j|}
+    $$
+
+    Parameters
+    ----------
+    n_samples: int
+        Number of samples in the design matrix.
+    n_features: int
+        Number of features in the design matrix.
+    corr: float
+        Correlation $\\rho$ between successive features. The element $C_{i, j}$
+        in the correlation matrix will be $\\rho^{|i-j|}$. This parameter
+        should be selected in $[0, 1[$.
+    random_state : int | RandomState instance | None (default)
+        Determines random number generation for centroid initialization and
+        random reassignment. Use an int to make the randomness deterministic.
+
+    Returns
+    -------
+    X: ndarray, shape (n_samples, n_features)
+        A design matrix with Toeplitz covariance.
+    """
+    if rho >= 1 or rho < 0:
+        raise ValueError("The correlation `rho` should be chosen in [0, 1[.")
+    rng = check_random_state(random_state)
+
+    sigma = np.sqrt(1 - rho * rho)
+    innovation = rng.randn(n_features + 1, n_samples)
```

Comment:
any reason to generate them all in one go instead of calling rng for every column generation ?

Reply:
This is faster to have only one call than multiple calls as it reduces the overhead of calling the blass I would say but I did not benchmark.

Reply:
It seems that this is not the case?!?
 
![image](https://user-images.githubusercontent.com/3321081/98584390-1bc9a180-22c6-11eb-8c39-9c8a83f23528.png)